crates/sui-move-build/src/lib.rs

```diff
@@ -907,3 +907,43 @@ pub fn check_invalid_dependencies(invalid: &BTreeMap<Symbol, String>) -> Result<
         error: error_messages.join("\n"),
     })
 }
+
+pub fn check_conflicting_addresses(
+    conflicting: &BTreeMap<Symbol, (ObjectID, ObjectID)>,
+    dump_bytecode_base64: bool,
+) -> Result<(), SuiError> {
+    if conflicting.is_empty() {
+        return Ok(());
+    }
+
+    let suffix = if conflicting.len() == 1 { "" } else { "es" };
+
+    let err_msg = format!("found the following conflicting published package address{suffix}:");
+    let suggestion_message =
+        "You may want to:
+ - delete the published-at address in the `Move.toml` if the `Move.lock` address is correct; OR
+ - update the `Move.lock` address to be the same as the `Move.toml`; OR
```

Comment:
I think there's never any reason to keep the `published-at` address, right?

Reply:
Also, it seems like we have enough information here to actually spit out suggestions that people can copy-paste in? I have a dream that someday in the far-far-far future there will be an autofix =)